### ISSUE TITLE
YOLOv8 on a one-class detection

### ISSUE THREAD

**Issue by Avv22** [ultralytics/ultralytics#13489]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have read the issue below:

https://github.com/ultralytics/ultralytics/issues/6183

Can you please provide me details if we can use YOLOv8 only for single class detection and if yes it would be that in the image it would only detect one class? What is the reason that we should do single class detection please?

**Comment by glenn-jocher** [comment]
@Avv22 hello!

Thank you for your question and for checking the existing issues and discussions before posting. Yes, you can certainly use YOLOv8 for single-class detection. Here are the details:

### Single-Class Detection with YOLOv8

YOLOv8 can be configured to detect only one class by setting up your dataset and training configuration accordingly. Here’s how you can do it:

1. **Dataset Preparation**: Ensure your dataset is labeled for only one class. If you are using a custom dataset, make sure the annotation files reflect this single class.

2. **Configuration**: When training your model, specify the `single_cls` argument to treat all classes as a single class. This can be done in both the CLI and Python API.

#### Example in Python:
```python
from ultralytics import YOLO

# Load a model
model = YOLO("yolov8n.yaml")  # build a new model from YAML
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)

# Train the model for single-class detection
results = model.train(data="path/to/your/dataset.yaml", epochs=100, imgsz=640, single_cls=True)
```

#### Example in CLI:
```bash
yolo detect train data=path/to/your/dataset.yaml model=yolov8n.pt epochs=100 imgsz=640 single_cls=True
```

### Why Use Single-Class Detection?

1. **Simplicity**: If your application only requires detecting one type of object, simplifying the model to focus on that single class can improve performance and reduce complexity.
2. **Performance**: Training a model to detect only one class can lead to faster inference times and potentially higher accuracy for that specific class, as the model is not distracted by other classes.
3. **Resource Efficiency**: Single-class models can be more lightweight, making them suitable for deployment in resource-constrained environments such as mobile or edge devices.

### Verification

Before we proceed further, could you please ensure the following:
- You are using the latest versions of `torch` and `ultralytics`. If not, please upgrade and try again.
- Provide a minimum reproducible code example if you encounter any issues. This helps us investigate and resolve the problem more efficiently. You can refer to our guide on creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Feel free to reach out if you have any more questions or need further assistance. Happy training! 🚀

**Comment by Avv22** [comment]
The only pre-trained model you have is yolov8n.pt ? Do you recommend other models for this one-class task or the one above is good? I thought you might have several versions.

**Comment by glenn-jocher** [comment]
Hello @Avv22,

Thank you for your question! Yes, we do have multiple pre-trained models available for YOLOv8, each tailored for different performance and size requirements. The `yolov8n.pt` model is the smallest and fastest, but you can also consider other variants depending on your specific needs:

- **YOLOv8s.pt**: Small model, slightly larger and more accurate than `yolov8n.pt`.
- **YOLOv8m.pt**: Medium model, offering a good balance between speed and accuracy.
- **YOLOv8l.pt**: Large model, more accurate but slower.
- **YOLOv8x.pt**: Extra-large model, the most accurate but also the slowest.

For single-class detection, any of these models can be used effectively. The choice depends on your application's requirements for speed and accuracy. Here's a quick comparison:

- **Speed**: `yolov8n.pt` > `yolov8s.pt` > `yolov8m.pt` > `yolov8l.pt` > `yolov8x.pt`
- **Accuracy**: `yolov8x.pt` > `yolov8l.pt` > `yolov8m.pt` > `yolov8s.pt` > `yolov8n.pt`

You can start with `yolov8n.pt` for faster iterations and then experiment with larger models if you need higher accuracy.

If you encounter any issues or need further assistance, please ensure you are using the latest versions of `torch` and `ultralytics`. If the issue persists, providing a minimum reproducible code example will help us investigate further. You can find more details on creating one [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Happy experimenting! 🚀

**Comment by MedPhysLab** [comment]
@glenn-jocher hi,  for single class detection, iin the yaml file, do we have to list only one class ( the one we want to detect) or keep  2 classes like
0 background
1 target_class?

**Comment by glenn-jocher** [comment]
Hello @MedPhysLab, for single-class detection, you should list only the target class in the `.yaml` file. There is no need to include a separate "background" class, as YOLO inherently treats everything not labeled as background. Let me know if you have further questions!

**Comment by MedPhysLab** [comment]
@glenn-jocher  sorry , i have another question: in the case of object detection with one class, in the labels corresponding to images where there is not the object what do i write? i cannot write class 0 since it's the class of the object.
